### PR TITLE
Jaxnav docs & improvements

### PR DESCRIPTION
**Pull request by amacrutherford**

### CODE REVIEW

File: jaxmarl/environments/jaxnav/README.md [hunk 1]
```diff
@@ -1,5 +1,69 @@
...
-MORE TO COME.
\ No newline at end of file
+## Environment Details
+
+JaxNav was first introduced in ["No Regrets: Investigating and Improving Regret Approximations for Curriculum Discovery"](https://www.arxiv.org/abs/2408.15099) with an in-detail specification given in the Appendix.
+
+### Map Types
+The default map is square robots of width 0.5m moving within a world with grid based obstacled, with cells of size 1m x 1m. Map cell size can be varied to produce obstacles of higher fidelty or robot strucutre can be changed into any polygon or a circle.
+
+We also include a map which uses polygon obstacles, but note we have not used this code is a while so there may well be issues with it.
```
Comment (benellis3) [inline jaxmarl/environments/jaxnav/README.md:12]: Typo: "is a while"

File: jaxmarl/environments/jaxnav/README.md [hunk 2]
```diff
@@ -1,5 +1,69 @@
...
+
+### Observation space
+By default, each robot recieves 200 range readings from a 360-degree arc centered on their forward axis. These range readings have a max range of 6m but no minimum range and are discritised with a resultion of 0.05 m. Alongside these range readings, each robot recieves their current linear and angular velocities along with the direction to their goal. Their goal direction is given by a vector in polar form where the distance is either the max lidar range if the goal is beyond their "line of sight" or the actual distance if the goal is within their lidar range. There is no communication between agents.
```
Comment (benellis3) [inline jaxmarl/environments/jaxnav/README.md:15]: "discritised" -> "discretised"

Comment (benellis3) [reply jaxmarl/environments/jaxnav/README.md:15]: Also "recieves" -> "receives"

File: jaxmarl/environments/jaxnav/jaxnav_env.py [hunk 3]
```diff
@@ -232,8 +228,9 @@ def step_env(
         # 2) Check collisions, goal and time
         old_goal_reached = agent_states.goal_reached
         old_move_term = agent_states.move_term
-        map_collisions = self._check_map_collisions(new_pos, new_theta, agent_states.map_data)*(1-agent_states.done).astype(bool)
-        agent_collisions = self._check_agent_collisions(jnp.arange(agent_states.pos.shape[0]), new_pos, agent_states.done)*(1- agent_states.done).astype(bool)
+        map_collisions = jax.vmap(self._map_obj.check_agent_map_collision, in_axes=(0, 0, None))(new_pos, new_theta, agent_states.map_data)*(1-agent_states.done).astype(bool)
+        # map_collisions = self._check_map_collisions(new_pos, new_theta, agent_states.map_data)*(1-agent_states.done).astype(bool)
```
Comment (benellis3) [inline jaxmarl/environments/jaxnav/jaxnav_env.py:232]: Do you need this code?

File: jaxmarl/environments/jaxnav/jaxnav_singletons.py [hunk 4]
```diff
@@ -708,11 +708,6 @@ def make_jaxnav_singleton_collection(collection_id: str, **env_kwargs) -> Tuple[
         "NarrowChicane2b",
         "Chicane4",
     ],
-    "new": [
```
Comment (benellis3) [inline jaxmarl/environments/jaxnav/jaxnav_singletons.py:711]: Why did you remove these tests? Ideally tests should be fixed unless you are removing a feature

File: jaxmarl/environments/jaxnav/maps/grid_map.py [hunk 5]
```diff
@@ -330,6 +342,75 @@ def _passable(grid, posa, posb):
         
     def scale_coords(self, x):
         return x / self.cell_size
+    
+    def dikstra_path(self, map_data, pos1, pos2):
```
Comment (benellis3) [inline jaxmarl/environments/jaxnav/maps/grid_map.py:346]: Did you implement dijkstra's algorithm in Jax 😆 

absolutely nuts good work

Comment (benellis3) [reply jaxmarl/environments/jaxnav/maps/grid_map.py:346]: Would be good to describe how this works in a high level comment

File: jaxmarl/environments/jaxnav/maps/grid_map.py [hunk 6]
```diff
@@ -569,6 +651,44 @@ def _checkPointRect(x, y, rx, ry):
         inside = _checkPolyWithinRect(sides, grid_idx[1], grid_idx[0])
         return inside & map_grid[grid_idx[0], grid_idx[1]] & valid_idx
     
+    @partial(jax.jit, static_argnums=[0])
+    def check_all_agent_agent_collisions(self, agent_positions: chex.Array, agent_theta: chex.Array, agent_coords=None) -> chex.Array:
```
Comment (benellis3) [inline jaxmarl/environments/jaxnav/maps/grid_map.py:655]: Also would be good to provide more detail for people (like) me who have no idea what the separating axis theorem is.

**Review by benellis3 — APPROVED** [review]
Looks good to me -- left some typos and suggestions but feel free to ignore them.

File: jaxmarl/environments/jaxnav/jaxnav_singletons.py (hunk 4 above)
Comment (amacrutherford) [reply jaxmarl/environments/jaxnav/jaxnav_singletons.py:711]: oh these were just used during development so tidying them into a clearer set

File: jaxmarl/environments/jaxnav/maps/grid_map.py (hunk 5 above)
Comment (amacrutherford) [reply jaxmarl/environments/jaxnav/maps/grid_map.py:346]: at 11pm on a sunday yes haahah, Reviewer 4 asked for environment metrics..!

File: jaxmarl/environments/jaxnav/maps/grid_map.py (hunk 6 above)
Comment (amacrutherford) [reply jaxmarl/environments/jaxnav/maps/grid_map.py:655]: yea very fair point :)

File: jaxmarl/environments/jaxnav/README.md (hunk 2 above)
Comment (amacrutherford) [reply jaxmarl/environments/jaxnav/README.md:15]: the dyslexia do hit hard

File: jaxmarl/environments/jaxnav/jaxnav_env.py (hunk 3 above)
Comment (amacrutherford) [reply jaxmarl/environments/jaxnav/jaxnav_env.py:232]: nope :)
